apollo-runtime/src/main/java/com/apollographql/android/impl/ResponseBodyConverter.java

```diff
@@ -53,6 +59,7 @@
         }
       }
       jsonReader.endObject();
+      this.cache.cacheStore().merge(responseNormalizer.records());
```

Comment:
> is to either store all, or none of the records from a response

Completely agree.  And that what I actually meant by transaction, if we failed to parse then don't store at all. But at the same time we need to make sure that `responseNormalizer` won't crash parsing network response

>I also don't love that the ResponseBodyConverter is saving the records at all

can we  pass the instance of normalizer in constructor and move logic of merging to `RealApolloCall`?